tensorly/metrics/entropy.py

```diff
@@ -0,0 +1,33 @@
+from .. import backend as T
+from ..cp_tensor import CPTensor
+from ..cp_tensor import cp_normalize
+import numpy as np
+
+# Author: Taylor Lee Patti <taylorpatti@g.harvard.edu>
+
+
+def vonNeumann_entropy(tensor, offset=1e-12):
+    """Returns the von Neumann entropy of the tensor.
+
+    Parameters
+    ----------
+    tensor : (matrix or CP tensor)
+        Data structure
+    offset : (float)
+        Minimum value at which to not consider eigenvalue zero to avoid singularities.
+
+    Returns
+    -------
+    float
+    """
+    if isinstance(tensor, CPTensor):
```

Comment:
I'm wondering whether it would be worth having an explicit cp_von_neumann_entropy?

Reply:
So you would advise against autodetect for users? Or do you mean just define the main function in terms of the general, cp, and soon tt version? The later would lead to more repetitious code but

Reply:
Right -- the issue is that now the API for the class is not fixed and mostly experimental (something we need to work on). So far I have a function cp_do_something for most operations that users can call explicitly, though I'm not sure it's the best way either. 

In general now, all cp_tensors are in the form (weights, factors) so the cp_von_neumann_entropy could be:
```
def cp_neumann_entropy(cp_tensor):
    weights, factors = cp_normalize(cp_tensor)
    ...
```
which also will allow tuples, not just CPTensors

As a minor side note, would it make sense to name it just `neumann_entropy`?

Reply:
Okay, I get you, for consistency. I'll separate them then.

How about cp_vonneumann_entropy? I think the "von" makes the function's purpose clear - to my background at least.

Reply:
I agree, let's go with `vonneumann_entropy`